data_procesor/database_manipulating_class.py

```diff
@@ -0,0 +1,102 @@
+import sqlite3
+from datetime import datetime
+
+from constans import url, emoji
+from utilities.log_class import LogClass
+
+
+class DatabaseManipulating:
+    log: LogClass
+    connection: sqlite3.Connection
+    cursor: sqlite3.Cursor
+    flag: set()
+
+    def __init__(self, log):
+        self.log = log
+
+    def create_database(self):
+        try:
+            self.connection = sqlite3.connect(url.DATABASE_URI)
+            self.cursor = self.connection.cursor()
+            self.log.info(f"connected to SQLite successfully {emoji.TASK_SUCCSEEDED}")
+        except sqlite3.OperationalError as oe:
+            self.log.critical(f"Failed to connect to database {emoji.TASK_FAILED}."
+                              f" Operational error: {oe}")
+        except sqlite3.DatabaseError as de:
+            self.log.critical(f"Failed to connect to database {emoji.TASK_FAILED}. "
+                              f"Database error: {de}")
+        except Exception as e:
+            self.log.critical(f"failed to connect to database {emoji.TASK_FAILED}. The error: {e}")
+
+    def input_data_into_database_table(self, data):
+        try:
+            for hour, all_minutes in data:
+                for minutes in all_minutes:
+                    self.cursor.execute('INSERT INTO my_table (hour, minutes) VALUES (?, ?)',
+                                        (hour, minutes))
+            self.connection.commit()
+            self.log.info(f"Inserted data into database successfully {emoji.INFO} ")
+        except sqlite3.Error as e:
+            self.log.critical(f"failed to insert data into database{emoji.TASK_FAILED}. Error: {e}")
+
+    def close_database(self):
+        self.connection.close()
+        self.log.debug(f"database closed {emoji.TASK_SUCCSEEDED}")
+
+    def create_database_table(self):
+        self.connection = sqlite3.connect(url.DATABASE_URI)
+        self.cursor = self.connection.cursor()
+        try:
+            self.cursor.execute('''CREATE TABLE my_table (
+                                        id INTEGER PRIMARY KEY AUTOINCREMENT,
+                                        hour INTEGER,
+                                        minutes TEXT
+                                    )''')
+            self.log.debug(f"created new data base {emoji.TASK_SUCCSEEDED}")
+        except sqlite3.Error as e:
+            self.log.debug(e)
+            self.cursor.execute('DROP table my_table')
+            self.cursor.execute('''CREATE TABLE my_table (
+                                        id INTEGER PRIMARY KEY AUTOINCREMENT,
+                                        hour INTEGER,
+                                        minutes TEXT
+                                    )''')
+            self.log.debug(f"my_table was replaced with new table "
+                           f"under same name{emoji.TASK_SUCCSEEDED}")
```

Comment:
ar teisingai suprantu, tu cia parasei jeigu nepavyksta sukurti table, tu dropini table? Ar reikia to? nes jeigu dropinti pries kuriant nauja kiekviena karta tai ne sitoje vietoje tai reiktu daryti, gali gal placiau apie sita vieta :D

Reply:
aptarta gyvai

Reply:
yeee tavo budas veikia so went with that